netket/abstract_variational_driver.py

```diff
@@ -142,24 +142,30 @@ def advance(self, steps=1):
 
     def run(
         self,
-        output_prefix,
         n_iter,
+        out=None,
         obs=None,
-        save_params_every=50,
-        write_every=50,
-        step_size=1,
         show_progress=True,
+        save_params_every=50,  # for default logger
+        write_every=50,  # for default logger
+        step_size=1,  # for default logger
+        output_prefix=None,  # TODO: deprecated
     ):
         """
         Executes the Monte Carlo Variational optimization, updating the weights of the network
         stored in this driver for `n_iter` steps and dumping values of the observables `obs`
-        in the output. The output is a json file at `output_prefix`, overwriting files with
-        the same prefix.
+        in the output `logger`. If no logger is specified, creates a json file at `output_prefix`,
+        overwriting files with the same prefix.
+
+        !! Compatibility v2.1
+            Before v2.1 the order of the first two arguments, `n_iter` and `output_prefix` was
+            reversed. The reversed ordering will still be supported until v3.0, but is deprecated.
 
         Args:
+            :n_iter: the total number of iterations
             :output_prefix: The prefix at which json output should be stored (ignored if logger
               is provided).
-            :n_iter: the total number of iterations
+            :logger: a logger object to be used to log data
```

Comment:
```suggestion
            :out: A logger object to be used to store simulation log and data.
                If this argument is a string, it will be used as output prefix for the standard JSON logger.
```

Reply:
I had missed that one. It's now committed.